packages/bot/src/scratch/utils/active-symbols.js

```diff
@@ -0,0 +1,151 @@
+import PendingPromise from './pending-promise';
```

Comment:
@aaron-binary should we keep them in scratch? i mean is it gonna be used anywhere else?

Reply:
Good point, I've moved the `PendingPromise` class to `src/utils` so we can re-use it anywhere else.